src/BuildScriptGenerator/ExternalSdkProvider.cs

```diff
@@ -0,0 +1,255 @@
+        }
+      }
+
+      return false;
+    }
+
+    /// <inheritdoc />
+    public async Task<bool> RequestBlobAsync(string platformName, string blobName)
+    {
+      // Get the expected checksum for the SDK
+      var expectedChecksum = await this.GetChecksumForVersionAsync(platformName, blobName);
+      if (expectedChecksum == null)
+      {
+        this.logger.LogError("Failed to get checksum for blob : platform {platformName}, blobName {blobName}. Skipping download of blob.", platformName, blobName);
+        return false;
+      }
+
+      if (this.CheckLocalCacheForBlob(platformName, blobName, expectedChecksum))
+      {
+        return true;
+      }
+      else
+      {
+        try
+        {
+          var request = new SdkProviderRequest
+          {
+            PlatformName = platformName,
+            BlobName = blobName,
+            UrlParameters = new Dictionary<string, string>(),
+          };
+
+          var response = await this.SendRequestAsync(request);
+          var filePath = Path.Combine(ExternalSdksStorageDir, platformName, blobName);
+
+          if (response && File.Exists(filePath))
+          {
+            this.logger.LogInformation("Successfully requested blob for platform {platformName}, blobName {blobName}, available at {FilePath}", platformName, blobName, filePath);
+          }
+          else
+          {
+            this.logger.LogError("Failed to get blob for platform {platformName}, blobName {blobName}", platformName, blobName);
+            return false;
+          }
+        }
+        catch (Exception ex)
+        {
+          this.logger.LogError(ex, "Error requesting blob for platform {platformName} blobName {blobName} from external provider.", platformName, blobName);
+          return false;
+        }
+      }
+
+      // Verify checksum
+      var blobPath = Path.Combine(ExternalSdksStorageDir, platformName, blobName);
+      var actualChecksum = this.GetSHA512Checksum(blobPath);
+      if (string.Equals(actualChecksum, expectedChecksum, StringComparison.OrdinalIgnoreCase))
+      {
+          this.logger.LogInformation("Downloaded blob checksum verified successfully for platform {platformName}, blobName {blobName}", platformName, blobName);
+          return true;
+      }
+      else
+      {
+          this.logger.LogError("Checksum verification failed for downloaded blob: platform {platformName}, blobName {blobName}, actual checksum {actualChecksum} , expected checksum {expectedChecksum}", platformName, blobName, actualChecksum, expectedChecksum);
+          return false;
+      }
+    }
+
+    private async Task<bool> SendRequestAsync(SdkProviderRequest request)
+    {
+      using var socket = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified);
```

Comment:
We can reuse socket or dispose it at the end.

Reply:
here the socket would be disposed at the end 
here I was just using the `using` declatarion without braces which works the same way